sortinghat/core/models.py

```diff
@@ -183,3 +185,66 @@ class MatchingBlacklist(ModelBase):
 
     class Meta:
         db_table = 'matching_blacklist'
+
+
+class Transaction(ModelBase):
+    ADD_ID = 'add_identity'
+    DELETE_ID = 'delete_identity'
+    UPDATE_PROFILE = 'update_profile'
+    MOVE_ID = 'move_identity'
+    ADD_ORG = 'add_organization'
+    DELETE_ORG = 'delete_organization'
+    ADD_DOMAIN = 'add_domain'
+    DELETE_DOMAIN = 'delete_domain'
+    ENROLL = 'enroll'
+    WITHDRAW = 'withdraw'
+    MERGE_IDENTITIES = 'merge_identities'
+
+    NAME_CHOICES = ((ADD_ID, 'add_identity'), (DELETE_ID, 'delete_identity'),
+                    (UPDATE_PROFILE, 'update_profile'), (MOVE_ID, 'move_identity'),
+                    (ADD_ORG, 'add_organization'), (DELETE_ORG, 'delete_organization'),
+                    (ADD_DOMAIN, 'add_domain'), (DELETE_DOMAIN, 'delete_domain'),
+                    (ENROLL, 'enroll'), (WITHDRAW, 'withdraw'), (MERGE_IDENTITIES, 'merge_identities'))
+
+    tuid = CharField(max_length=MAX_SIZE_CHAR_FIELD, primary_key=True)
+    name = CharField(max_length=MAX_SIZE_CHAR_FIELD, choices=NAME_CHOICES, null=False)
+    timestamp = CreationDateTimeField()
+
+    class Meta:
+        db_table = 'transactions'
+        ordering = ('timestamp', 'tuid')
+
+    def __str__(self):
+        return '%s - %s' % (self.tuid, self.name)
+
+
+class Operation(ModelBase):
+    ADD = 'add'
+    DELETE = 'delete'
+    UPDATE = 'update'
+    UUID = 'uuid'
+    UID = 'uid'
+    ORG = 'org'
+    DOMAIN = 'domain'
+    ENROLLMENT = 'enrollment'
+    PROFILE = 'profile'
+    BLACKLIST = 'blacklist'
+
+    OP_TYPE_CHOICES = ((ADD, 'add'), (DELETE, 'delete'), (UPDATE, 'update'))
+    ENTITY_CHOICES = ((UUID, 'unique_identity'), (UID, 'identity'), (ORG, 'organization'),
+                      (DOMAIN, 'domain'), (ENROLLMENT, 'enrollment'), (PROFILE, 'profile'),
+                      (BLACKLIST, 'blacklist_entry'))
+
+    ouid = CharField(max_length=MAX_SIZE_CHAR_FIELD, primary_key=True)
+    type = CharField(max_length=MAX_SIZE_CHAR_FIELD, choices=OP_TYPE_CHOICES, null=False)
```

Comment:
Please check `null` fields because I think by default `null=False` is the defauld behaviour.

Reply:
You are right, I confirm `null=False` is the default behaviour: https://docs.djangoproject.com/en/2.2/ref/models/fields/#django.db.models.Field.null .